pkg/adapter/kubernetesevents/adapter.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetesevents
+
+import (
+	"context"
+	"github.com/knative/pkg/logging"
+	"github.com/knative/pkg/signals"
+	"go.uber.org/zap"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/knative/pkg/cloudevents"
+	corev1 "k8s.io/api/core/v1"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	eventType = "dev.knative.k8s.event"
+)
+
+type Adapter struct {
+	// Namespace is the kubernetes namespace to watch for events from.
+	Namespace string
+	// SinkURI is the URI messages will be forwarded on to.
+	SinkURI string
+
+	ctx context.Context
+}
+
+// Run starts the kubernetes receive adapter and sends events produced in the
+// configured namespace and send them to the sink uri provided.
+func (a *Adapter) Run(ctx context.Context) {
```

Comment:
I recommend changing this signature to conform to the [`Runnable` interface](https://github.com/kubernetes-sigs/controller-runtime/blob/81f67a05853ea208098ffb64ad2862a8892771ba/pkg/manager/manager.go#L130-L135), i.e. `Start(<-chat struct{}) error`.

The benefit of this is that the global code in this function (setting up signal handling, exiting the process) can be handled entirely in `main`, making this function more testable and easier to use in combination with other things.

A minor benefit is that controller-runtime's Manager could be brought in to take care of things like creating a stop channel, setting up signal handlers, and creating a kubernetes client.

The context argument doesn't seem to be used at the moment (The context passed in from main is `context.Background()`) so I don't think it will be difficult to remove.